core/service/manager.go

```diff
@@ -18,47 +18,64 @@
 package service
 
 import (
-	"crypto/x509/pkix"
-
 	log "github.com/cihub/seelog"
-	"github.com/mysteriumnetwork/go-openvpn/openvpn"
-	"github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/state"
-	"github.com/mysteriumnetwork/go-openvpn/openvpn/tls"
 	"github.com/mysteriumnetwork/node/communication"
-	"github.com/mysteriumnetwork/node/core/ip"
-	"github.com/mysteriumnetwork/node/core/location"
+	nats_dialog "github.com/mysteriumnetwork/node/communication/nats/dialog"
+	nats_discovery "github.com/mysteriumnetwork/node/communication/nats/discovery"
 	"github.com/mysteriumnetwork/node/discovery"
 	"github.com/mysteriumnetwork/node/identity"
+	identity_registry "github.com/mysteriumnetwork/node/identity/registry"
 	identity_selector "github.com/mysteriumnetwork/node/identity/selector"
+	"github.com/mysteriumnetwork/node/logconfig"
 	"github.com/mysteriumnetwork/node/metadata"
-	"github.com/mysteriumnetwork/node/nat"
 	dto_discovery "github.com/mysteriumnetwork/node/service_discovery/dto"
-	openvpn_discovery "github.com/mysteriumnetwork/node/services/openvpn/discovery"
-	"github.com/mysteriumnetwork/node/session"
 )
 
-// Manager represent entrypoint for Mysterium service with top level components
+const logPrefix = "[service-manager] "
+
+// Service interface represents pluggable Mysterium service
+type Service interface {
+	Start(providerID identity.Identity, dialogWaiter communication.DialogWaiter) (dto_discovery.ServiceProposal, error)
+	Wait() error
+	Stop() error
+}
+
+// NewManager creates new instance of pluggable services manager
+func NewManager(
+	networkDefinition metadata.NetworkDefinition,
+	identityLoader identity_selector.Loader,
+	signerFactory identity.SignerFactory,
+	identityRegistry identity_registry.IdentityRegistry,
+	service Service,
+	discoveryService *discovery.Discovery,
+) *Manager {
+	logconfig.Bootstrap()
+
+	return &Manager{
+		identityLoader: identityLoader,
+		dialogWaiterFactory: func(myID identity.Identity) communication.DialogWaiter {
```

Comment:
Is there a better naming then `myID`? 
It's not clear whose id should be there.

Reply:
Renamed to `providerID`

Reply:
Well it depends, You can create Dialog at consumer side and wait for dialogs. It's ID of the one who waits for dialogs